tests/test_bls.py

```diff
@@ -85,3 +88,26 @@ def test_g2_subgroup_check():
 
     gasCost = test_bls.isOnSubgroupG2DLZZGasCost(format_G2(valid_G2))
     print("G2 subgroup check DLZZ", gasCost)
+
+def test_aggregation():
+    test_bls = accounts[0].deploy(TestBLS)
+
+    secret_key1 = 123
+    secret_key2 = 456
+
+    public_key1 = get_public_key(secret_key1)
+    public_key2 = get_public_key(secret_key2)
+    agg_public_key = aggregate_public_keys([public_key1, public_key2])
+    
+    data = encode_hex("fooooo")
+    message_solc = tuple(test_bls.hashToPoint(data))
+    message = parse_solc_G1(message_solc)
+
+    sig1 = sign(message, secret_key1)
+    sig2 = sign(message, secret_key2)
+    agg_sig = aggregate_signatures([sig1, sig2])
+
+    agg_pubkey_solc = format_G2(agg_public_key)
+    agg_sig_solc = format_G1(agg_sig)
+
+    assert test_bls.verifySingle(agg_sig_solc, agg_pubkey_solc, message_solc)
```

Comment:
Do you think we can verify this with the `verifyMultiple` function? The function takes individual public keys as arguments and aggregates them on chain, preventing the rouge key attack.

Reply:
Sure, I kept "verifySignle" just to test aggregate_public_keys